pkg/client/unversioned/request.go

```diff
@@ -585,8 +594,17 @@ func (r *Request) Watch() (watch.Interface, error) {
 	if client == nil {
 		client = http.DefaultClient
 	}
+
+	time.Sleep(time.Duration(backoff.CalculateBackoff(url)) * time.Second)
 	resp, err := client.Do(req)
 	updateURLMetrics(r, resp, err)
+	if r.baseURL != nil {
+		if err != nil {
+			backoff.UpdateBackoff(r.baseURL.Path, err, 0)
```

Comment:
I actually went back and forth on this and wound up picking string just for simplicity wrt underlying libs used... But I think you're right for the final patch a URL will be More precise